flight_computer/packets.py

```diff
@@ -139,127 +88,113 @@ def data(self, data: dict) -> None:
     @property
     def raw_data(self) -> bytearray:
         """Get raw values as a bitarray"""
-        data_list = []
+        val_list = []
         for field, vals in self.data.items():
             for val in vals:
-                data_list.append(val)
-        arr = pack(data_list, self._fmt)
-        return arr
+                val_list.append(val)
+
+        def pack() -> bytearray:
+            packed_data = bytearray()
+            bool_list = []
+            for fmt, val in zip(self._fmt, val_list):
+                if fmt is type_name_to_fmt["bool"]:
+                    bool_list.append(val)
+                else:
+                    packed_data += struct.pack(ENDIAN + fmt, val)
+
+            packed_bools = bytearray()
+            j = 0
+            bool_byte = 0
+            for b in bool_list:
+                bool_byte |= (b << j)
+                j = (j + 1) % 8
+                if j is 0:
+                    packed_bools += struct.pack(ENDIAN + U_8, bool_byte)  # unsigned char to represent 8 bits!
+                    bool_byte = 0
+            if j is not 0 and len(bool_list) > 0:
+                packed_bools += struct.pack(ENDIAN + U_8, bool_byte)
+            packed_data += packed_bools
+            return packed_data
+
+        return pack()
 
     @raw_data.setter
     def raw_data(self, arr: bytearray) -> None:
-        """Set raw data from bitarray"""
+        """Set raw data from bytearray"""
+        val_list = []
+
+        def unpack():
+            """return a list of values parsed according to the packets fmt string"""
+            k_arr = 0
+            j_bit = 0
+            for fmt in self._fmt:
+                if fmt is not type_name_to_fmt["bool"]:
+                    raw_bytes = arr[k_arr: k_arr + fmt_to_byte_size[fmt]]
+                    val = struct.unpack(ENDIAN + fmt, raw_bytes)[0]
+                    val_list.append(val)
+                    k_arr += fmt_to_byte_size[fmt]
+                else:
+                    # get truth value at jth position at the kth byte in the byte array
+                    bitmask = 1 << j_bit
+                    raw_byte = arr[k_arr:k_arr + 1]
+                    val = bitmask & struct.unpack(ENDIAN + U_8, raw_byte)[0]
+                    val = bool(val)
+                    val_list.append(val)
+                    j_bit = (j_bit + 1) % 8
+                    if j_bit is 0:
+                        k_arr += 1
+            return val_list
```

Comment:
I feel like having the method in a method is a bit confusing. Maybe just make this a private class method (prefix it with an underscore)?